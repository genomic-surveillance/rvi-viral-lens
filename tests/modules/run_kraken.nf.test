nextflow_process {

    name "Test Process run_kraken"
    script "modules/run_kraken.nf"
    process "run_kraken"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                test_loc="${projectDir}/tests/test_data"
            }
            process {
                
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    params.meta, 
                    [
                        file("${params.test_loc}/test_input_fastqs/test_1.fastq.gz"),
                        file("${params.test_loc}/test_input_fastqs/test_2.fastq.gz")
                    ]
                    )
                input[1] = file("${params.test_loc}/test_kraken_databases/minimal")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
