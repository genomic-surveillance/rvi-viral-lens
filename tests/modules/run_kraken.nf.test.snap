{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.kraken.output:md5,2aa58b378359d98568407dda371ea60b",
                        [
                            "test.class_seqs_1.fq:md5,54558937e0ddd9aa3fedab9025050518",
                            "test.class_seqs_2.fq:md5,831d5601d5b964c2482327e6860be3fa"
                        ],
                        [
                            "test.unclass_seqs_1.fq:md5,d8c82cc39046db60237e01edabfcddd6",
                            "test.unclass_seqs_2.fq:md5,8149ffc6c83395c20de3b23823f80295"
                        ],
                        "test.kraken_report.txt:md5,4cee02e460ea9ab3ea84c8d9ef32dbbd"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-14T09:51:28.499556641"
    }
}